dockers/actions-plan-preview/github.go

```diff
@@ -123,3 +127,109 @@ func getPullRequest(ctx context.Context, client *github.Client, owner, repo stri
 	pr, _, err := client.PullRequests.Get(ctx, owner, repo, prNum)
 	return pr, err
 }
+
+type issueCommentQuery struct {
+	ID     githubv4.ID
+	Author struct {
+		Login githubv4.String
+	}
+	Body        githubv4.String
+	IsMinimized githubv4.Boolean
+}
+
+type issueCommentsQuery struct {
+	Nodes    []issueCommentQuery
+	PageInfo struct {
+		HasPreviousPage githubv4.Boolean
+		StartCursor     githubv4.String
+	}
+}
+
+type pullRequestCommentQuery struct {
+	Repository struct {
+		PullRequest struct {
+			Comments issueCommentsQuery `graphql:"comments(last: 100, before: $commentsCursor)"`
+		} `graphql:"pullRequest(number: $prNumber)"`
+	} `graphql:"repository(owner: $repositoryOwner, name: $repositoryName)"`
+}
+
+var errNotFound = errors.New("not found")
+
+// find the latest plan preview comment in the specified issue
+// if there is no plan preview comment, return errNotFound err
+func findLatestPlanPreviewComment(ctx context.Context, client *githubv4.Client, owner, repo string, prNumber int) (*issueCommentQuery, error) {
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	variables := map[string]interface{}{
+		"repositoryOwner": githubv4.String(owner),
+		"repositoryName":  githubv4.String(repo),
+		"prNumber":        githubv4.Int(prNumber),
+		"commentsCursor":  (*githubv4.String)(nil),
+	}
+
+	var q pullRequestCommentQuery
+	for ctx.Err() == nil {
```

Comment:
Could you tell me why do we need to do a loop here?

Reply:
This is for when there are so many comments on pr that the first fetch doesn't find the comment (this is rare case I think)

Reply:
I see your point. But I think it is quite rare since we are fetching 100 last comments.
So for the sake of simplicity, I prefer to remove the loop. What do you think?

Reply:
I agree. It is rare and simplicity is better.
I'll remove loop. Thank you!